kahypar/partition/partitioner.h

```diff
@@ -177,7 +178,18 @@ inline void Partitioner::configurePreprocessing(const Hypergraph& hypergraph,
     if (density < 0.75) {
       context.preprocessing.community_detection.edge_weight = LouvainEdgeWeight::degree;
     } else {
-      context.preprocessing.community_detection.edge_weight = LouvainEdgeWeight::uniform;
+      bool contains_large_he = false;
+      for ( const HyperedgeID he : hypergraph.edges() ) {
+        if ( hypergraph.edgeSize(he) > LARGE_HE_THRESHOLD ) {
+          contains_large_he = true;
+          break;
+        }
+      }
+      if ( density < 2 && contains_large_he ) {
```

Comment:
Would you mind adding a brief comment in the code here to explain the rationale behind this?

Reply:
done